mobile/native/src/ln_dlc/mod.rs

```diff
@@ -821,6 +825,79 @@ pub fn collaborative_revert_channel(
     Ok(())
 }
 
+fn update_state_after_collab_revert(
+    sub_channel: SubChannel,
+    execution_price: Decimal,
+) -> Result<()> {
+    let positions = db::get_positions()?;
+
+    let position = match positions.first() {
+        Some(position) => {
+            tracing::info!("Channel is reverted before the position got closed successfully.");
+            position
+        }
+        None => {
+            tracing::info!("Channel is reverted before the position got opened successfully.");
+            if let Some(order) = db::maybe_get_order_in_filling()? {
+                order::handler::order_failed(
+                    Some(order.id),
+                    FailureReason::ProposeDlcChannel,
+                    anyhow!("Order failed due collab revert of the channel"),
+                )?;
+            }
+            return Ok(());
+        }
+    };
+
+    let filled_order = match order::handler::order_filled() {
+        Ok(order) => order,
+        Err(_) => {
+            let order = Order {
+                id: uuid::Uuid::new_v4(),
+                leverage: position.leverage,
+                quantity: position.quantity,
+                contract_symbol: position.contract_symbol,
+                direction: position.direction.opposite(),
+                order_type: OrderType::Market,
+                state: OrderState::Filled {
+                    execution_price: execution_price.to_f32().expect("to fit into f32"),
+                },
+                creation_timestamp: OffsetDateTime::now_utc(),
+                order_expiry_timestamp: OffsetDateTime::now_utc(),
+                reason: OrderReason::Expired,
+                stable: position.stable,
+            };
+            db::insert_order(order)?;
+            event::publish(&EventInternal::OrderUpdateNotification(order));
+            order
+        }
+    };
+
+    position::handler::update_position_after_dlc_closure(Some(filled_order))?;
+
+    match db::delete_positions() {
+        Ok(_) => {
+            event::publish(&EventInternal::PositionCloseNotification(
+                ContractSymbol::BtcUsd,
+            ));
+        }
+        Err(error) => {
+            tracing::error!("Could not delete position : {error:#}");
+        }
+    }
+
+    let mut sub_channel = sub_channel;
+    sub_channel.state = SubChannelState::OnChainClosed;
+
+    let node = NODE.try_get().context("failed to get ln dlc node")?;
```

Comment:
Might be best to do this at the very beginning of the function to no do any of the other updates if we really can't get the node.